### PR TITLE
Expand entrypoint docs

### LINKED ISSUE #58: Entrypoints - Additional Info
I think the [entrypoints page](https://cosmwasm-docs.vercel.app/core/entrypoints) would be a good place to explain what `DepsMut`, `Env` and `MessageInfo` are for. We don't need to go into a lot of detail, but a short sentence and a docs.rs link for each would be nice.

The docs should also explain that the `InstantiateMsg` / `ExecuteMsg` / ... are defined by the user and how to do that (e.g. explain `#[cw_serde]`).
A similar explanation for the returned error type would be good. Right now all of the examples use `StdResult`, but you're totally free to use any type that implements `ToString`.

### PR DESCRIPTION
**Pull request by aumetra**
Closes #58

### CODE REVIEW

File: src/pages/core/entrypoints.mdx [hunk 1]
```diff
@@ -14,6 +14,16 @@ these entrypoints, each one different from the last.
 In this section we want to give you a quick overview over all the entrypoints
 and when they are called.
 
+<Callout>
+  In our examples, we will always refer to `StdResult` as the error type.
```
Comment (uint) [inline src/pages/core/entrypoints.mdx]: I find this line slightly confusing since `StdError` is the error type. Maybe rephrase?

**Review by uint — APPROVED** [review]
Looks solid! Just one suggestion, take it or leave it.

File: src/pages/core/entrypoints.mdx (hunk 1 above)
Comment (aumetra) [reply src/pages/core/entrypoints.mdx]: You're right. I was thinking wrong there. Gonna update that
